File: laravel/app/app/Models/Job.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Database\Eloquent\Relations\HasMany;
use Illuminate\Database\Eloquent\Casts\Attribute;
use Illuminate\Database\Eloquent\Factories\HasFactory;
use Orchid\Screen\AsSource;

class Job extends Model
{
    use HasFactory, AsSource;

    protected $fillable = [
        'user_id',
        'name',
        'url',
        'cron_expression',  
        'method',
        'payload',
        'headers',        
        'is_active',        
        'last_run_at',     
        'next_run_at',
    ];

    protected $casts = [
        'payload' => 'array',
        'headers' => 'array',           
        'is_active' => 'boolean',
        'last_run_at' => 'datetime',   
        'next_run_at' => 'datetime',
    ];

    public function user(): BelongsTo
    {
        return $this->belongsTo(User::class);
    }

    public function execution(): HasMany
    {
        return $this->hasMany(Execution::class)->latest('started_at');
    }
}

File: go-scheduler/internal/models/job.go
package models

import (
    "encoding/json"
    "time"
)

type Job struct {
    ID             int             `json:"id"`
    UserID         int             `json:"user_id"`
    Name           string          `json:"name"`
    URL            string          `json:"url"`
    Method         string          `json:"method"`
    CronExpression string          `json:"cron_expression"`
    Payload        json.RawMessage `json:"payload"`
    Headers        json.RawMessage `json:"headers"`
    IsActive       bool            `json:"is_active"`
    LastRunAt      *time.Time      `json:"last_run_at"`
    NextRunAt      *time.Time      `json:"next_run_at"`
}

type DueJobsResponse struct {
    Jobs []Job `json:"jobs"`
}

type QueueMessage struct {
    JobID     int       `json:"job_id"`
    URL       string    `json:"url"`
    Method    string    `json:"method"`
    Payload   string    `json:"payload"`
    Headers   string    `json:"headers"`
    Timestamp time.Time `json:"timestamp"`
}